### PR TITLE
Refactor time series pipeline code via time series pipeline base class

### LINKED ISSUE #2614: Refactor time series pipeline inheritance
``TimeSeriesClassificationPipeline`` inherits from ``ClassificationPipeline`` and ``TimeSeriesRegressionPipeline`` inherits from ``RegressionPipeline``. However, these classes override almost all of the methods in these base classes (predict, score, predict_proba, fit). In addition, ``TimeSeriesClassificationPipeline`` and ``TimeSeriesRegressionPipeline`` have suspiciously similar code for each of these methods.

I think we could refactor these classes s.t. we have a TimeSeriesBasePipeline class which has predict, score, predict_proba, fit as defined in both classes. The classification time series pipelines will have to do some extra encoding, but at least we could share the logic for padding.

If we want to take advantage of any functionality in our normal pipelines, we can use mixins, as we already are doing for `TimeSeriesBinaryClassificationPipeline` via `BinaryClassificationPipelineMixin`.

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #2614

### CODE REVIEW

**Review by chukarsten — APPROVED** [review]
LGTM!

File: evalml/pipelines/time_series_regression_pipeline.py [hunk 1]
```diff
@@ -10,9 +7,7 @@
 )
 
 
-class TimeSeriesRegressionPipeline(
-    RegressionPipeline, metaclass=TimeSeriesPipelineBaseMeta
-):
+class TimeSeriesRegressionPipeline(TimeSeriesPipelineBase):
```
Comment (freddyaboulton) [inline evalml/pipelines/time_series_regression_pipeline.py:10]: Can you please mark that `TimeSeriesRegressionPipeline` does not inherit from `RegressionPipeline` as a breaking change?

**Review by freddyaboulton — APPROVED** [review]
Thank you @angela97lin !

**Review by jeremyliweishih — APPROVED** [review]
looks GREAT to me!

File: evalml/pipelines/time_series_regression_pipeline.py (hunk 1 above)
Comment (angela97lin) [reply evalml/pipelines/time_series_regression_pipeline.py:10]: sure thing 😁

**Pull request merged**